packages/lwc/index.d.ts

```diff
@@ -191,42 +191,64 @@ declare module 'lwc' {
 
     /**
      * Decorator factory to wire a property or method to a wire adapter data source
+     * Use generic types to allow type checking for wire adapters
+     * Default all the generic types to any to maintain backward compatibility
      * @param adapter the adapter used to provision data
      * @param config configuration object for the adapter
      */
-    export function wire(
-        adapter: WireAdapterConstructor | LegacyWireAdapterConstructor,
-        config?: WireConfigValue
+    export function wire<
```

Comment:
Could you add an example of what a fully typed wire `@wire` would look like with this change?